records/2022-04-07.reliability.service.configuration.md

```diff
@@ -0,0 +1,36 @@
+# Improve Notify service configuration reliability
+
+Date: 2022-04-07
+
+## Status
+
+**In review.**
+
+## Context
+
+The Notify API, Admin and Document Download API are configured using environment variables.  These environment variables are stored in encrypted `.env` files and Kubernetes manifests.  Overtime, misconfigurations have occurred because of inconsistencies in how the String values of the environment variables are converted to Boolean and Integer values.  
+
+The goal of this solution is to improve the reliability of service configuration by consistently setting configuration values across services and logging the service configuration when it starts so that it can be validated. 
+
+## Proposed solution
+
+There are two proposed solutions for this ADR.
+
+### Log configuration on service startup
+
+Update the Notify API, Admin and Document Download API to log their configuration values on startup.  This will allow us to validate that the running services have the expected configuration values.  Sensitive values will be redacted in the logs to prevent accidental exposure.
```

Comment:
spaces nitpicking

```suggestion
Update the Notify API, Admin and Document Download API to log their configuration values on startup. This will allow us to validate that the running services have the expected configuration values. Sensitive values will be redacted in the logs to prevent accidental exposure.
```